### PR TITLE
SDK-190 [Java SDK] Schema 2.2 support

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-190](https://incountry.atlassian.net/browse/SDK-190)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: src/main/java/com/incountry/residence/sdk/dto/search/FindFilterBuilder.java [hunk 1]
```diff
@@ -42,46 +42,46 @@ public FindFilterBuilder limitAndOffset(int limit, int offset) throws StorageCli
         return this;
     }
 
-    public FindFilterBuilder stringKeyEq(StringField filed, String... keys) throws StorageClientException {
+    public FindFilterBuilder keyEq(StringField filed, String... keys) throws StorageClientException {
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/residence/sdk/dto/search/FindFilterBuilder.java:45]: filed -> field 
typo

File: src/main/java/com/incountry/residence/sdk/dto/search/FindFilterBuilder.java [hunk 2]
```diff
@@ -42,46 +42,46 @@ public FindFilterBuilder limitAndOffset(int limit, int offset) throws StorageCli
...
-    public FindFilterBuilder stringKeyNotEq(StringField filed, String... keys) throws StorageClientException {
-        filter.setStringFilter(filed, new FilterStringParam(keys, true));
+    public FindFilterBuilder keyEq(NumberField field, Long... keys) throws StorageClientException {
+        filter.setNumberFilter(field, new FilterNumberParam(keys));
         return this;
     }
 
-    public FindFilterBuilder numberKeyEq(NumberField field, Long... keys) throws StorageClientException {
-        filter.setNumberFilter(field, new FilterNumberParam(keys));
+    public FindFilterBuilder keyNotEq(StringField filed, String... keys) throws StorageClientException {
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/residence/sdk/dto/search/FindFilterBuilder.java:55]: typo

File: src/main/java/com/incountry/residence/sdk/tools/crypto/CryptoManager.java [hunk 3]
```diff
@@ -235,4 +235,8 @@ private String decryptBase64(String cipherText) {
         byte[] decodedBytes = Base64.getDecoder().decode(cipherText);
         return new String(decodedBytes, CHARSET);
     }
+
+    public boolean isUsePTEncryption() {
+        return usePTEncryption;
+    }
 }
```
[the remark this answers is not in the document]
Comment (VladimirLevitskiy) [reply src/main/java/com/incountry/residence/sdk/tools/crypto/CryptoManager.java:242]: It is not necessary for Java

File: src/test/java/com/incountry/residence/sdk/crypto/AllSdkEncryptionTest.java [hunk 4]
```diff
@@ -21,12 +21,12 @@
     private static final String ENV_ID = "InCountry";
     private static final String PASSWORD = "password";
     private static final String RESPONSE = "{\n" +
-            "    \"key\": \"976143aa1fd12b9ad7449fd9d3a6d25347d71b890b49d4fb5c738e798238865f\",\n" +
-            "    \"body\": \"2:IGJNCmV+RXZydaPxDjjhZ80/6aZ2vcEUZ2GuOzKgVSSdM6gYf5RPgFbyLqv+7ihz0CpYFQQWf9xkIyD/u3VYky8dWLq+NXcE2xYL4/U7LqUZmJPQzgcQCABYQ/8vOvUEcrfOAwzGjR6etTp1ki+79JmCEZFSNqcDP1GZXNLFdLoSUp1X2wVlH9ukhJ4jrE0cKDrpJllswRSOz0BhS8PA/73KNKwo718t7fPWpUm7RkyILwYTd/LpPvpXMS6JypEns8fviOpbCLQPpZNBe6zpwbFf3C0qElHlbCyPbDyUiMzVKOwWlYFpozFcRyWegjJ42T8v52+GuRY5\",\n" +
```
Comment (RommeroInCountry) [inline src/test/java/com/incountry/residence/sdk/crypto/AllSdkEncryptionTest.java:25]: Please keep old data as well. It'll be useful to test against "old meta". Simply add new record alongside with new body

File: src/main/java/com/incountry/residence/sdk/dto/search/StringField.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.incountry.residence.sdk.dto.search;
+
+public enum StringField {
+    VERSION,
```
[the remark this answers is not in the document]
Comment (VladimirLevitskiy) [reply src/main/java/com/incountry/residence/sdk/dto/search/StringField.java:4]: After discussion, we decided to save current code until PoP API will be refactored (support NOT filter for numeric fields)

File: src/test/java/com/incountry/residence/sdk/crypto/AllSdkEncryptionTest.java (hunk 4 above)
Comment (VladimirLevitskiy) [reply src/test/java/com/incountry/residence/sdk/crypto/AllSdkEncryptionTest.java:25]: Fixed.

**Review by RommeroInCountry — APPROVED** [review]
LGTM

**Review by RommeroInCountry — APPROVED** [review]
LGTM

**Review by robert-par — APPROVED** [review]

**Pull request merged**